src/Propel/Generator/Behavior/Validate/ValidateBehavior.php

```diff
@@ -237,6 +229,26 @@ protected function addLoadValidatorMetadataMethod()
         return $this->renderTemplate('objectLoadValidatorMetadata', ['constraints' => $constraints]);
     }
 
+    /**
+     * Propel constraints are preferred over Symfony ones.
+     *
+     * @param array $properties
+     *
+     * @return string
+     */
+    protected function getClassConstraint(array $properties)
+    {
+        if (class_exists("Propel\\Runtime\\Validator\\Constraints\\".$properties['validator'], true)) {
+            return "Propel\\Runtime\\Validator\\Constraints\\".$properties['validator'];
+        }
+
+        if (class_exists("Symfony\\Component\\Validator\\Constraints\\".$properties['validator'], true)) {
+            return "Symfony\\Component\\Validator\\Constraints\\".$properties['validator'];
```

Comment:
It might be worthwhile to assign these to two variables to reduce the possibility of making a mistake by 2 very tiny points.

```suggestion
        $propelValidator = "Propel\\Runtime\\Validator\\Constraints\\".$properties['validator'];
        if (class_exists($propelValidator, true)) {
            return $propelValidator;
        }
        
        $symfonyValidator = "Symfony\\Component\\Validator\\Constraints\\".$properties['validator'];
        if (class_exists($symfonyValidator, true)) {
            return symfonyValidator;
```

Also, what if we composed this behaviour (or injected it) instead?

```
        $constraintValidatorTargets = [
            "Propel\\Runtime\\Validator\\Constraints\\".$properties['validator'],
            "Symfony\\Component\\Validator\\Constraints\\".$properties['validator']
        ];
```